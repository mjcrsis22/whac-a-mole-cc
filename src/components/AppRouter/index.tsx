import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';
import { APP_ROUTES } from '../../constants/routes';
import HomeView from '../../views/Home';
import GameView from '../../views/Game';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route index element={<Navigate to={APP_ROUTES.HOME} replace />} />
        <Route path={APP_ROUTES.HOME} element={<HomeView />} />
        <Route path={APP_ROUTES.GAME} element={<GameView />} />
        <Route path="*" element={<Navigate to="/" replace />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
