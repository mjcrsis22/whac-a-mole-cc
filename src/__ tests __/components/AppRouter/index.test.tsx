import { render } from '@testing-library/react';
import AppRouter from '../../../components/AppRouter';

jest.mock('../../../views/Home', () => () => <>View Home</>);
jest.mock('../../../views/Game', () => () => <>View Game</>);

describe('AppRouter component', () => {
  test('Renders correctly', () => {
    const { container } = render(<AppRouter />);
    expect(container).toMatchSnapshot();
  });
});
