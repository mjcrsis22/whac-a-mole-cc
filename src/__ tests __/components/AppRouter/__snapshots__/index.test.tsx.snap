// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter component Renders correctly 1`] = `
<div>
  View Home
</div>
`;
